pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -489,21 +503,59 @@ func (bfs *Service) IsActive() bool {
 	return bfs.isActive.Load()
 }
 
+// retry function with exponential backoff.
+func (bfs *Service) retry(action func() error) error {
+	var (
+		err     error
+		backoff = initialBackoff
+	)
+
+	for range maxRetries {
+		if err = action(); err == nil {
+			return nil
+		}
+
+		if bfs.ctx.Err() != nil {
```

Comment:
not sure that the second check is needed

Reply:
This check is not needed. If context is cancelled, then `action()` will return an error.